src/app/3d/qgs3daxis.cpp

```diff
@@ -0,0 +1,646 @@
+/***************************************************************************
+  qgs3daxis.cpp
+  --------------------------------------
+  Date                 : March 2022
+  Copyright            : (C) 2022 by Jean Felder
+  Email                : jean dot felder at oslandia dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgs3daxis.h"
+
+#include <Qt3DCore/QTransform>
+#include <Qt3DExtras/QConeMesh>
+#include <Qt3DExtras/QCylinderMesh>
+#include <Qt3DExtras/QPhongMaterial>
+#include <Qt3DRender/qcameralens.h>
+#include <Qt3DRender/QCameraSelector>
+#include <Qt3DRender/QClearBuffers>
+#include <Qt3DRender/QLayer>
+#include <Qt3DRender/QLayerFilter>
+#include <Qt3DRender/QPointLight>
+#include<ctime>
+
+#include <Qt3DRender/QAttribute>
+
+Qgs3DAxis::Qgs3DAxis( Qt3DExtras::Qt3DWindow *parentWindow, Qt3DCore::QEntity *parent3DScene, QgsCameraController *cameraCtrl, const Qgs3DMapSettings *map )
+  : QObject( parentWindow ), mParentWindow( parentWindow ), mParentCamera( cameraCtrl->camera() ),
+    mCylinderLength( 40.0f ), mAxisViewportSize( 4.0 * mCylinderLength ),
```

Comment:
please move all the initializiers into the header, c++11 style.

Reply:
ok